biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -196,7 +196,7 @@ boolean refresh() throws Exception {
 		}
 		return descriptors.values()
 				.stream()
-				.filter(descriptor -> {
+				.anyMatch(descriptor -> {
```

Comment:
This is wrong since anyMatch short circuits the stream. ALL elements in the stream must be processed. This is why count is used!

Reply:
The side effects are important.

Reply:
If side effects are important, we shouldn't be using a stream operation that is intended to be stateless. This should probably just be a foreach loop. I'll update the PR today.